### PR TITLE
BUG: fix the value error when using read_csv with "," as decimal point and pd.Float64Dtype()

### LINKED ISSUE #52086: BUG: support decimal keyword for Float64Dtype in read_csv
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [x] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
import io
pd.read_csv(io.StringIO('id\n"1,5"\n'), dtype={'id':pd.Float64Dtype()}, sep=';', decimal=',')
```
or
```python
import pandas as pd
pd.read_csv("./test.csv", dtype={'id':pd.Float64Dtype()}, sep=';', decimal=',')
```
where test.csv looks like
```
id
1,5
```

### Issue Description

I have a semicolon-separated CSV file with a bunch of floats that have decimal separation with comma.
For that setup, when I specify `dtype` for that column as `Float64Dtype()`, it fails.

When I specify `float` from python, it works.
When I use and specify `.` as a decimal separator, it works.

### Expected Behavior

The listed example should parse and result in a float `1.5`

### Installed Versions

<details>

/home/robbert/.local/lib/python3.10/site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")

INSTALLED VERSIONS
------------------
commit           : 2e218d10984e9919f0296931d92ea851c6a6faf5
python           : 3.10.6.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.19.0-35-generic
Version          : #36~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Fri Feb 17 15:17:25 UTC 2
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.5.3
numpy            : 1.23.5
pytz             : 2022.1
dateutil         : 2.8.1
setuptools       : 65.5.1
pip              : 23.0.1
Cython           : None
pytest           : 7.2.1
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : 1.1
pymysql          : None
psycopg2         : 2.9.5
jinja2           : 3.1.2
IPython          : 8.6.0
pandas_datareader: None
bs4              : 4.11.1
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : 3.1.1
pandas_gbq       : None
pyarrow          : 10.0.1
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
snappy           : None
sqlalchemy       : 1.4.46
tables           : None
tabulate         : 0.9.0
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : 2022.7

</details>

### PR DESCRIPTION
**Pull request by hxy450**
- [x] closes #52086 
- [x] [Tests added and passed]
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations]
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/_libs/parsers.pyx [hunk 1]
```diff
@@ -1201,6 +1201,9 @@ cdef class TextReader:
                         result, dtype=dtype, true_values=true_values,
                         false_values=false_values)
                 else:
+                    if chr(self.parser.decimal) != ".":
```
Comment (phofl) [inline pandas/_libs/parsers.pyx:1204]: This costs to much performance, I think that we have to fix this somewhere else

**Comment by hxy450** [comment]
This is my first contribution so I'm still trying to find my feet. 
The current problem is after this function passes the array containing strings to _from_sequence_of_strings(), we lose track of what decimal point we should use. As a result, a latter function floatify() cannot parse the string correctly since they are trying to parse the string with ".". Is it ok if I add additional parameters to all the following functions to indicate this decimal point? Or is that inefficient too?

**Comment by phofl** [comment]
This needs restructuring of the casting logic. This is non-trivial.

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**